### PR TITLE
Use fmt to format messages

### PR DESCRIPTION
**Pull request by ranisalt**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

[fmt](https://fmt.dev) is a library that provides fast and safe string formatting, with a syntax similar to `printf` (or `str.format` for my fellow Pythonists).

This pull request adds it as a dependency, and also changes some usages of stringstreams for formatting where `fmt::format` would suit objectively better. There are many more places there `fmt::format` would fit (just grep for `std::ostringstream` to find), but I can't do that much alone 😛 

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

**Review by MillhioreBT — APPROVED** [review]

File: src/game.cpp [hunk 1]
```diff
@@ -4004,12 +4006,7 @@ bool Game::combatChangeHealth(Creature* attacker, Creature* target, CombatDamage
 		realHealthChange = target->getHealth() - realHealthChange;
 
 		if (realHealthChange > 0 && !target->isInGhostMode()) {
-			std::stringstream ss;
-
-			ss << realHealthChange << (realHealthChange != 1 ? " hitpoints." : " hitpoint.");
-			std::string damageString = ss.str();
-
-			std::string spectatorMessage;
+			auto damageString = fmt::format("{:d} hitpoint{:s}.", realHealthChange, realHealthChange != 1 ? "s" : "");
```
Comment (MillhioreBT) [inline src/game.cpp:4009]: image
this point is out of place.

**Review by MillhioreBT — CHANGES_REQUESTED** [review]

File: src/game.cpp (hunk 1 above)
Comment (ranisalt) [reply src/game.cpp]: Oops.

**Comment by yamaken93** [comment]
Signed

**Comment by nekiro** [comment]
This is great. I think you missed item descriptions though (or you wanted to miss it :D)

**Review by nekiro — APPROVED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -68,6 +69,7 @@ target_link_libraries(tfs PRIVATE
         Boost::system
         Boost::filesystem
         Boost::iostreams
+	fmt::fmt
```
Comment (DSpeichert) [inline CMakeLists.txt:72]: Some whitespace trolololo here

**Review by DSpeichert — CHANGES_REQUESTED** [review]
Looks much better indeed. Just some tab/space issue for consistency.

File: CMakeLists.txt (hunk 2 above)
Comment (ranisalt) [reply CMakeLists.txt:72]: Shit. I can never get CMakeLists.txt proper formatting

File: CMakeLists.txt [hunk 3]
```diff
@@ -39,6 +39,7 @@ if (CryptoPP_FOUND)  # vcpkg-provided cmake package called CryptoPP
 else()
     find_package(Crypto++ REQUIRED)
 endif ()
+find_package(fmt REQUIRED)
```
Comment (djarek) [inline CMakeLists.txt:42]: This should probably look for 7.1.x or higher.

**Review by DSpeichert — APPROVED** [review]

File: CMakeLists.txt (hunk 3 above)
Comment (ranisalt) [reply CMakeLists.txt:42]: Ubuntu only ships 6.1, and 18.04 ships 4.0

I hate this trash...

**Review by MillhioreBT — DISMISSED** [review]

File: CMakeLists.txt (hunk 3 above)
Comment (djarek) [reply CMakeLists.txt:42]: AFAIK, 7.x has `std::fmt` compatibility, previous versions had issues (because the proposal was changed in standardization).

Comment (ranisalt) [reply CMakeLists.txt]: I couldn't find documentation on that, but I locked to the version that the build job runs (6.1.2). I have 7.1.3 here

**Review by DSpeichert — APPROVED** [review]

**Comment by nekiro** [comment]
I'm curious, how's the performance of fmt compared to stringstream?

**Comment by ranisalt** [comment]
> I'm curious, how's the performance of fmt compared to ostringstream?

[It should be orders of magnitude faster](https://www.zverovich.net/2020/06/13/fast-int-to-string-revisited.html), especially when converting numbers to text. For just concatenating strings, it should be at least 2x faster.

[It is also forward-compatible with `std::format`](https://www.zverovich.net/2019/07/23/std-format-cpp20.html)

**Comment by yamaken93** [comment]
Just for the record(std::format vs fmt in general): https://www.reddit.com/r/cpp/comments/hm7gz0/fmt_70_released_with_code_size_performance_and/fx3wd58/?utm_source=reddit&utm_medium=web2x&context=3

**Review by yamaken93 — APPROVED** [review]

**Comment by yamaken93** [comment]
I have reviewed it again and it seems pretty solid for me.

**Review by MillhioreBT — APPROVED** [review]
I have tried it again and LGTM

**Review by DSpeichert — APPROVED** [review]

**Pull request merged**